manager/src/main/java/io/gingersnapproject/RuleResource.java

```diff
@@ -23,19 +21,34 @@ public class RuleResource {
    @Inject
    QueryHandler queryHandler;
 
+   @Inject
+   RuleManager ruleManager;
+
    @GET
    @Operation(summary = "Retrieve a cache entry associated with the provided rule and key")
    @Path("/{rule}/{key}")
    @Produces(MediaType.APPLICATION_JSON)
    public Uni<String> get(String rule, String key) {
-      return maps.get(rule, key);
+      checkRuleExists(rule);
+      Uni<String> uni = maps.get(rule, key);
+      if (uni instanceof UniItem<?> uniItem)  {
+         if (uniItem.getItem() == null)
+            throw new NotFoundException();
```

Comment:
Looks good to me, I wonder if we should cache this exception. It is probably a somewhat common case to have a miss, so I think it might be worth it.